webapp/src/services/api.ts

```diff
@@ -195,25 +195,17 @@ export const api = createApi({
     updateDataActions: build.mutation<
       UtterancePatch[],
       {
-        persistentIds: UtterancePatch["persistentId"][];
-        newValue: DataAction;
+        utterancePatch: UtterancePatch[];
       } & GetUtterancesQueryState
     >({
-      queryFn: async ({ jobId, datasetSplitName, persistentIds, newValue }) =>
+      queryFn: async ({ utterancePatch, ...utteranceQuery }) =>
```

Comment:
Most of the fields from `utteranceQuery` are not allowed for the `PATCH`. They are used in the `args` for `api.util.updateQueryData("getUtterances", args, ...` to perform the right optimistic update.
This should be reverted to using only `jobId` and `datasetSplitName`, as well as `utterancePatch`/`body`, and ignore the rest.
```suggestion
      queryFn: async ({ jobId, datasetSplitName, body }) =>
```